### PR TITLE
♻️ Splits models_library.services

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ♻️ Splits `models_library.services` into more manageable sub-modules. This is the rationale
  - the module/submodules follow a structure similar to e.g. [aiohttp](https://github.com/aio-libs/aiohttp/tree/master/aiohttp)
      -  `models_library.services` acts as the **main module API**
      -  `models_library.services_*` act as **submodules**. Can be either imported directly and/or some functionality is exposed to the main module API 
  - highlights differences between different service metadata (SMD) models and how they are composed. We have SMD models for:
     1. image (static)
     2. database (user-defined, editable)
     3. webserver API schema ( GET/PATCH )
     4. catalog API schema ( GET/PATCH )
- 🔨 new `make doc-uml` on any package to produce UML diagrams. Based in [pyreverse](https://pylint.pycqa.org/en/latest/pyreverse.html) which comes with pylint.

## Related issue/s

- resolves https://github.com/ITISFoundation/osparc-simcore/issues/3486
- preparation for  https://github.com/ITISFoundation/osparc-issues/issues/1404

## How to test

- Using `doc-uml` to trace code inhertance/composition relations between models
   -  `cd packages/models-library` (or any other package/service)
   - `make -n  doc-uml`
   - `(export DOC_UML_PATH_SUFFIX="services*"; export DOC_UML_CLASS=models_library.api_schemas_catalog.services.ServiceGet; make doc-uml)`
   - will produce a UML diagram

## Dev-ops checklist
None

### CODE REVIEW

File: packages/models-library/src/models_library/basic_regex.py [hunk 1]
```diff
@@ -75,3 +75,5 @@
     r"(?::(?P<docker_tag>[\w][\w.-]{0,127}))?"
     r"(?P<docker_digest>\@sha256:[a-fA-F0-9]{32,64})?$"
 )
+
+PROPERTY_KEY_RE = r"^[-_a-zA-Z0-9]+$"  # TODO: PC->* it would be advisable to have this "variable friendly" (see VARIABLE_NAME_RE)
```
Comment (sanderegg) [inline packages/models-library/src/models_library/basic_regex.py:79]: forgotten todo, and it is actually json key friendly (at least preventing invalid ones, I guess there are plenty of RE available for these)

File: packages/models-library/src/models_library/projects_nodes.py [hunk 2]
```diff
@@ -29,7 +30,7 @@
 )
 from .projects_nodes_ui import Position
 from .projects_state import RunningState
-from .services import PROPERTY_KEY_RE, ServiceKey, ServiceVersion
+from .services import ServiceKey, ServiceVersion
 
 # NOTE: WARNING the order here matters
```
Comment (sanderegg) [inline packages/models-library/src/models_library/projects_nodes.py:35]: when I see the positioning of this warning, I think this lost its initial location. but I guess this has to do with the union types below..
There are also a FIXME and a TODO below (line 41  and 52)

File: packages/models-library/src/models_library/services_authoring.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        description="Link to the status",
+        examples=[
+            "https://travis-ci.org/ITISFoundation/osparc-simcore 'State of CI: build, test and pushing images'",
+            "https://coveralls.io/github/ITISFoundation/osparc-simcore?branch=master 'Test coverage'",
+            "https://itisfoundation.github.io/",
+        ],
+    )
+
+    class Config:
+        extra = Extra.forbid
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_authoring.py:32]: so do we want to follow the nice extra philosophy of not caring if there are additional keys and ignore them?

File: packages/models-library/src/models_library/services_constants.py [hunk 4]
```diff
@@ -11,3 +10,40 @@
...
+    r"(?P<name>[a-z0-9-_]+[a-z0-9])$"
+)
+COMPUTATIONAL_SERVICE_KEY_FORMAT: Final[str] = "simcore/services/comp/{service_name}"
+
+
+LATEST_INTEGRATION_VERSION: Final[str] = "1.0.0"
+
+
+#
+ANY_FILETYPE: Final[str] = "data:*/*"
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_constants.py:49]: these are actually the only constants in this module (with line 35)... it is a bit ill named no? or maybe the regexes should move to basic_regex

File: packages/models-library/src/models_library/services_enums.py [hunk 5]
```diff
@@ -40,3 +40,24 @@ def comparison_order() -> dict["ServiceState", int]:
...
+
+
+class ServiceType(str, Enum):
+    COMPUTATIONAL = "computational"
+    DYNAMIC = "dynamic"
+    FRONTEND = "frontend"
+    BACKEND = "backend"
+
+
+# TODO: create a flags enum that accounts for every column
```
Comment (sanderegg) [inline packages/models-library/src/models_library/services_enums.py:52]: forgotten todo?

**Review by sanderegg — APPROVED** [review]
all good, you have a service_runtime.py that is empty

File: packages/models-library/src/models_library/projects_nodes.py (hunk 2 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/projects_nodes.py:35]: This is for the union, yes

File: packages/models-library/src/models_library/services_authoring.py (hunk 3 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/services_authoring.py:32]: IMO when it is an output, it should be strict (we define exactly what is provided) and when it is an input, it should be flexible (we take what we need and ignore the rest)

File: packages/models-library/src/models_library/services_io.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+    # value
+    file_to_key_map: dict[FileName, ServicePortKey] | None = Field(
+        None,
+        alias="fileToKeyMap",
+        description="Place the data associated with the named keys in files",
+        examples=[{"dir/input1.txt": "key_1", "dir33/input2.txt": "key2"}],
+    )
+
+    # TODO: should deprecate since content_schema include units
```
Comment (GitHK) [inline packages/models-library/src/models_library/services_io.py:81]: maybe this message can be addressed

File: packages/models-library/src/models_library/services_io.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            "contentSchema": port_schema,
+        }
+
+
+class ServiceInput(BaseServiceIOModel):
+    """
+    Metadata on a service input port
+    """
+
+    # TODO: should deprecate since content_schema include defaults as well
```
Comment (GitHK) [inline packages/models-library/src/models_library/services_io.py:134]: same here

**Review by GitHK — APPROVED** [review]
👍

**Review by bisgaard-itis — APPROVED** [review]
Looks good to me. Thanks for the effort!

**Pull request merged**